### PR TITLE
Tasks run only async

### PR DESCRIPTION
**Pull request by IvanPombo**
This PR sets that all our simulations will run async moving forward. There are still ways to block the task with the `task.wait()` and combine it with `task.sync_context()` in order to retrieve the features of blocking and killing locally. 

This was tested with all scenarios that are in the documentation and flowed perfectly. To check on the tasks I used the CLI just done by @luispcunha - that is why it is merged here, which means that those blocking calls don't matter as much now either way. 

Please verify this PR thoroughly for all small fixes and things you don't understand. Run your own scripts with it to test it as well. If all is okay, it still goes with the release!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -144,12 +144,9 @@ The user must specify the input directory containing the files to run the simula
 
 ## Running multiple simulations in parallel
 
-Up until now, all simulations were run synchronously which gives feedback while the simulation is running until it finishes. However, this is not always the best option, for example, when launching two simulations it implies that they run one after the other. This
-is not optimal for when users need to launch hundreds of simulations.
+**Inductiva API** allows you to launch multiple simulation in parallel. After a simulation is submitted, the client unblocks - simulations run asynchronously - and can run other tasks. Users are not limited to wait for the simulation to finish.
```
Comment (cristianacarpinteiro) [inline README.md:147]: multiple simulations*

Comment (cristianacarpinteiro) [reply README.md:147]: Maybe rephrase to: After a simulation is submitted, the client unblocks - simulations run asynchronously - and can execute other commands.

Because tasks is something we use a lot to refer to simulations.

Comment (cristianacarpinteiro) [reply README.md:147]: are not limited by waiting for ...

**Review by cristianacarpinteiro — APPROVED** [review]
Everything looks okay to me! Left some minor comments

File: README.md (hunk 1 above)
Comment (begaiym-k) [inline README.md:147]: Users don't know what the client is, so other option: "After submitting a simulation, users are free to perform other tasks since simulations run independently in the background."?

File: inductiva/fluids/wind_tunnel/README.md [hunk 2]
```diff
@@ -56,7 +56,7 @@ The simulation parameters available for the user to configure are:
 - `num_iterations`: Set the maximum number of iterations for the iterative algorithm to converge.
 - `resolution`: Controls the resolution of the meshing that is done prior to the simulation. The higher the resolution, the finer the meshing. Available options: "high", "medium", "low".
 
-Moreover, the hardware and interaction are configured with the usual general parameters - `machine_group`, `run_async`, `n_cores`.
+Moreover, the hardware and interaction are configured with the usual general parameters - `machine_group`, `n_cores`.
```
Comment (hpenedones) [inline inductiva/fluids/wind_tunnel/README.md:59]: is n_cores still there?

File: inductiva/molecules/protein_solvation/README.md [hunk 3]
```diff
@@ -58,7 +58,7 @@ The user can further define the following simulation parameters:
 - `n_steps_min`: Number of steps for energy minimization. Default value is 5000.
 - `ignore_warnings`: Whether to ignore warnings during grompp (gromacs preprocessor) step. If set to False, the simulation will fail if there are warnings. If True, the simulation will run and ignore the warnings, but the results may have inaccuracies. Use with caution. 
 
-Moreover, the hardware and interaction are configured with the usual general parameters - `machine_group`, `run_async`. Launching a simulation returns a task object, which can be used to verify the status of the simulation, get the simulation outputs and access post-processing tools. See more in [Tasks](inductiva/README.md).
+Moreover, the hardware can be configured through the `machine_group` object. Launching a simulation returns a task object, which can be used to verify the status of the simulation, get the simulation outputs and access post-processing tools. See more in [Tasks](inductiva/README.md).
```
Comment (begaiym-k) [inline inductiva/molecules/protein_solvation/README.md:61]: `inductiva.resources.MachineGroup`*

File: inductiva/fluids/wind_tunnel/README.md (hunk 2 above)
Comment (IvanPombo) [reply inductiva/fluids/wind_tunnel/README.md:59]: Yes, for OpenFOAM we need extra care due to changing the commands based on the n_cores. This is an extra step that needs a more careful consideration to move to the backend, since it interferes with the low-level simulation as well.

File: inductiva/tasks/README.md [hunk 4]
```diff
@@ -1,57 +1,21 @@
 # Tasks
 
 
-The **Inductiva API** client provides the ability to run complex simulations and contains easy-to-use tools to manage them. Each simulation submitted - with `scenario.simulate()` or `simulation.run()` - returns a `Task` object that contains methods to manage the simulation.
-
+The **Inductiva API** client provides the ability to run complex simulations and contains easy-to-use tools to manage them. Each simulation submitted - with `scenario.simulate()` or `simulation.run()` - runs asynchronously and the client unblocks returning a `Task` object that contains methods to manage the simulation.
```
Comment (begaiym-k) [inline inductiva/tasks/README.md:4]: Again "the client".  "runs asynchronously and the Task object is returned"?

File: README.md (hunk 1 above)
Comment (IvanPombo) [reply README.md:147]: Wonderfull!

**Comment by IvanPombo** [comment]
@hpenedones One thing that I meant today is: Can we create an extra branch for development - e.g., `main-dev`? This change iterates on the documentation and removes the async everywhere. However, this change is not reflected on the PyPI, hence if merge to the `main` branch users will have incorrect information when checking this documentation and won't know what is going on. If Yes, then it would be good to change the defaults so that when opening a PR, it immediately compares with `main-dev` instead of `main`. Obviously, we can still merge to `main` bug fixes.

**Comment by hpenedones** [comment]
Yes, we can try to have a dev branch for this intermediate release. Please chat with the other developers to get organised around this. Thanks!

**Comment by IvanPombo** [comment]
> Yes, we can try to have a dev branch for this intermediate release. Please chat with the other developers to get organised around this. Thanks!

Would you be so kind to change the defaults of opening a PR directly to the branch named `development` whenever you can? :) I think neither me or @luispcunha as power to do that.

**Review by Joao97ribeiro — APPROVED** [review]
LGTM!
